core/state_transition.go

```diff
@@ -254,6 +254,7 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 		}
 		//if input is empty for the smart contract call, return
```

Comment:
Would be useful if you could a comment that this condition will occur when the node is not a participant to a private transaction.